index.js

```diff
@@ -97,84 +97,167 @@ const parseArgs = (
+        // e.g. '--foo'
+        optionKey = StringPrototypeSlice(arg, 2);
+        if (isExpectingValue(optionKey, options) && isOptionValue(nextArg)) {
+          // e.g. '--foo' 'bar'
+          optionValue = nextArg;
+          pos++;
+        }
       }
-
-    } else {
-      // Arguments without a dash prefix are considered "positional"
-      ArrayPrototypePush(result.positionals, arg);
+      storeOptionValue(optionKey, optionValue, options, result);
+      pos++;
+      continue;
     }
 
+    // Anything that did not get handled above is a positional.
+    ArrayPrototypePush(result.positionals, arg);
     pos++;
   }
 
   return result;
 };
 
+/**
+  * Determines if `arg` is a just a short option.
+  * @example '-f'
+  */
+function isLoneShortOption(arg) {
+  return arg.length === 2 &&
+    StringPrototypeCharAt(arg, 0) === '-' &&
+    StringPrototypeCharAt(arg, 1) !== '-';
+}
+
+/**
+  * Determines if `arg` is a short option group.
+  *
+  * See Guideline 5 of the [Open Group Utility Conventions](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html).
+  *   One or more options without option-arguments, followed by at most one
+  *   option that takes an option-argument, should be accepted when grouped
+  *   behind one '-' delimiter.
+  * @example
+  * isShortOptionGroup('-a', {}) // returns false
+  * isShortOptionGroup('-ab', {}) // returns true
+  * isShortOptionGroup('-fb', { withValue: ['f'] }) // returns false
+  * isShortOptionGroup('-bf', { withValue: ['f'] }) // returns true
+  */
+function isShortOptionGroup(arg, options) {
+  if (arg.length <= 2) return false;
+  if (StringPrototypeCharAt(arg, 0) !== '-') return false;
+  if (StringPrototypeCharAt(arg, 1) === '-') return false;
+
+  const onlyFlags = arg.slice(1, -1);
+  for (let index = 0; index < onlyFlags.length; index++) {
+    const optionKey = getOptionKey(StringPrototypeCharAt(onlyFlags, index));
+    if (isExpectingValue(optionKey, options)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+/**
+  * Determines if `arg` is a long option, which may have a trailing value.
+  * @example
+  * isLongOption('-a) // returns false
+  * isLongOption('--foo) // returns true
+  * isLongOption('--foo=bar) // returns true
+  */
+function isLongOption(arg) {
+  return arg.length > 2 && StringPrototypeStartsWith(arg, '--');
+}
+
+/**
+  * Expand known short options into long, otherwise return original.
+  * @example
+  * getOptionKey('f', { short: { f: 'file'}}) // returns 'file'
+  * getOptionKey('b', {}) // returns 'b'
+  * getOptionKey('long-option', {}) // returns 'long-option'
+  */
+function getOptionKey(option, options) {
+  if (option.length === 1 && options?.short?.[option]) {
+    return options.short[option]; // long option
+  }
+  return option;
+}
+
+/**
+ * Determines if the option is expecting a value.
+ */
+function isExpectingValue(optionKey, options) {
+  return options && options.withValue &&
```

Comment:
```suggestion
  return options && ArrayIsArray(options.withValue) &&
```
otherwise a truthy non-array could cause problems

Reply:
if we're sure it's an array when present, why would `options` be falsy?

Reply:
Could be missing with current validation: https://github.com/pkgjs/parseargs/blob/2ab5af3cafef37d2545de212033d9955aaaef48d/index.js#L87

(We currently validate, but not modify. Might be revamping configuration: #63)